cluster/gke/composition.yaml

```diff
@@ -1,39 +1,96 @@
 apiVersion: apiextensions.crossplane.io/v1
 kind: Composition
 metadata:
-  name: gke.gcp.platformref.crossplane.io
+  name: gke.gcp.platformref.upbound.io
   labels:
     provider: GCP
 spec:
   writeConnectionSecretsToNamespace: upbound-system
   compositeTypeRef:
-    apiVersion: gcp.platformref.crossplane.io/v1alpha1
+    apiVersion: gcp.platformref.upbound.io/v1alpha1
     kind: GKE
   resources:
-    - base:
-        apiVersion: container.gcp.crossplane.io/v1beta2
+    - name: service-account
+      base:
+        apiVersion: cloudplatform.gcp.upbound.io/v1beta1
+        kind: ServiceAccount
+      patches:
+        - fromFieldPath: spec.id
+          toFieldPath: metadata.name
+        - fromFieldPath: spec.id
+          toFieldPath: spec.forProvider.displayName
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.email
+          toFieldPath: status.gke.serviceAccount
+          policy:
+            fromFieldPath: Required
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.id
+          toFieldPath: status.gke.project
+          transforms:
+            - type: string
+              string:
+                type: Regexp
+                regexp:
+                  match: projects\/(.+)\/serviceAccounts\/.*
+                  group: 1
+          policy:
+            fromFieldPath: Required
+    - name: service-account-key
+      base:
+        apiVersion: cloudplatform.gcp.upbound.io/v1beta1
+        kind: ServiceAccountKey
+        spec:
+          forProvider:
+            serviceAccountIdSelector:
+              matchControllerRef: true
+      patches:
+        - fromFieldPath: spec.id
+          toFieldPath: metadata.name
```

Comment:
```suggestion
```

Seems like keys don't have names. Not patching it at all would be enough here to prevent conflicts I think.